tianshou/policy/modelfree/a2c.py

```diff
@@ -1,15 +1,27 @@
-from typing import Any, Dict, List, Optional, Type
+from typing import Any, Callable, Dict, List, Optional, cast
 
 import numpy as np
 import torch
 import torch.nn.functional as F
 from torch import nn
 
-from tianshou.data import Batch, ReplayBuffer, to_torch_as
+from tianshou.data import ReplayBuffer, to_torch_as
+from tianshou.data.types import BatchWithReturnsProtocol, RolloutBatchProtocol
 from tianshou.policy import PGPolicy
+from tianshou.policy.modelfree.pg import TDistParams
 from tianshou.utils.net.common import ActorCritic
 
 
+class BatchWithAdvantagesProtocol(BatchWithReturnsProtocol):
+    """Contains estimated advantages and values.
+
+    Returns are usually computed from GAE of advantages by adding the value.
+    """
+
+    adv: torch.Tensor
+    v_s: torch.Tensor
```

Comment:
move this to data/types.py?

Reply:
Done, don't know how it slipped through my search